### PR TITLE
Generate client queries based on datamodel

### PR DESCRIPTION
**Pull request by ankitjena**
## How it works
Generate queries and mutation with fragments according to config(which are enabled). #72 

### Mutation
```
import gql from "graphql-tag"
import { NoteFragment } from "../fragments/Note"

export const updateNote = gql`
  mutation updateNote($id: ID!, title: String!, description: String!, published: Boolean!) {
    updateNote(id: $id, input: {title: $title, description: $description, published: $published}) {4
      ...NoteFields
    }
  }

  ${NoteFragment}
`
```

### Query
```
import gql from "graphql-tag"
import { NoteFragment } from "../fragments/Note"

export const findAllNotes = gql`
  query findAllNotes {
    findAllNotes {
      ...NoteFields
    }
  }

  ${NoteFragment}
`
```
### Subscription
```
import gql from "graphql-tag"
import { CommentFragment } from "../fragments/Comment"

export const newComment = gql`
  subscription newComment {
    newComment {
      ...CommentFields
    }
  }

  ${CommentFragment}
`
```

### Fragment
```
import gql from "graphql-tag"

export const NoteFragment = gql`
  fragment NoteFields on Note {
    id
    title
    description
    published
  }
`
```

### CODE REVIEW

File: packages/graphback/src/resolvers/outputResolvers/resolverTemplate.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ const generateTypeResolvers = (context: TargetResolverContext, name: string): st
   let output = `${imports}`
 
   if(context.subscriptions.length) {
-    output += `\n\n enum Subscriptions {
+    output += `\n\nenum Subscriptions {
```
Comment (wtrocki) [inline packages/graphback/src/resolvers/outputResolvers/resolverTemplate.ts]: ??

File: packages/graphback-cli/src/templates/configTemplates.ts [hunk 2]
```diff
@@ -56,10 +56,20 @@ export const chooseDatabase = async(): Promise<string> => {
   return database
 }
 
+export const askForClient = async(): Promise<boolean> => {
+  const { client } = await ask({
+    type: 'confirm',
+    name: 'client',
+    message: 'Do you want to generate client?'
```
Comment (wtrocki) [inline packages/graphback-cli/src/templates/configTemplates.ts]:
```suggestion
    message: 'Do you want to generate client-side GraphQL queries?'
```

**Comment by wtrocki** [comment]
We need documentation and release notes. I will need to play with that much longer over the comming days.

File: packages/graphback/src/resolvers/outputResolvers/resolverTemplate.ts (hunk 1 above)
Comment (ankitjena) [reply packages/graphback/src/resolvers/outputResolvers/resolverTemplate.ts]: No its `\n` it left empty space so had indentation error.

**Comment by ankitjena** [comment]
> Good work. I think we might need to have a blog post about it.
> Please add docs and we are sorted

But we haven't integrated uniforms, shouldn't we download client template in init

**Comment by wtrocki** [comment]
@ankitjena No. Client side will be never focus of graphback, however we can deliver client side artifacts to work on clients. 

WE need documentation that our cli will generate queries in some folder. 
Then people can simple use `create-react-app`, but I'm working with uniforms team to pick that up and have `create-uniforms-app` that will be giving us end to end client support
thru thrid party.

I think it is important to be able to change folder of the client queries generation.

File: docs/clientsidequeries.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
+---
+id: clientqueries
+title: Client-side Queries
+---
+
+Additionaly graphback also allows users to generate client-side queries based on the data model.
```
Comment (wtrocki) [inline docs/clientsidequeries.md]:
```suggestion
Graphback allows users to generate client-side queries based on the data model.
```

File: docs/clientsidequeries.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+These can be used by client side frameworks like [uniform.tools](https://uniforms.tools) to create
```
Comment (wtrocki) [inline docs/clientsidequeries.md]: I would move this to separate . chapter here. 
Something like

## Autogenerating forms based on server-side schema

Comment (wtrocki) [reply docs/clientsidequeries.md]: Then we will collaborate with @radekme to get proper integration

Comment (wtrocki) [reply docs/clientsidequeries.md]: Then we will collaborate with @radekmie to get proper integration

File: docs/clientsidequeries.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+}
+```
+running `graphback generate` will generate all operations which are enabled in `config.json` or using `directives`.
+
+## Examples
+### Fragment
+```
+import gql from "graphql-tag"
+
+export const NoteFragment = gql`
```
Comment (wtrocki) [inline docs/clientsidequeries.md]: It is important that we done it this way.

Uniforms can base on NoteFragments to build forms input. Separate tooling I have build will apply common patterns - for example hide id field and generate labels based on the metadata provided in directives

File: docs/clientsidequeries.md (hunk 3 above)
Comment (wtrocki) [inline docs/clientsidequeries.md]: Missing info that users still need to follow up with any available clients like Apollo and use something like create-react-app to kickstart experience.

File: docs/clientsidequeries.md (hunk 4 above)
Comment (ankitjena) [reply docs/clientsidequeries.md]: Sure

**Pull request merged**